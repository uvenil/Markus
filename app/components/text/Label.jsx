'use strict';

import React from 'react';

export default class Label extends React.Component {
    constructor(props) {
        super(props);
    }

    render() {
        const theme     = this.props.theme === 'dark' ? require('../../theme.dark.json') : require('../../theme.light.json');
        const textColor = this.props.textColor ? this.props.textColor : theme.primaryTextColor;

        return (
            <span
                className={this.props.className}
                style={{ WebkitUserSelect : 'none', cursor : 'default', fontFamily : this.props.fontFamily, fontWeight : this.props.fontWeight, fontSize : this.props.textSize, color : textColor, pointerEvents : this.props.disabled ? 'none' : 'auto' }}>
                {this.props.children}
            </span>
        );
    }
}

Label.propTypes = {
    className  : React.PropTypes.string,
    fontFamily : React.PropTypes.string,
    fontWeight : React.PropTypes.oneOfType([ React.PropTypes.string, React.PropTypes.number ]),
    textSize   : React.PropTypes.string,
    textColor  : React.PropTypes.string,
    theme      : React.PropTypes.oneOf([ 'light', 'dark' ]),
    disabled   : React.PropTypes.bool
};

Label.defaultProps = {
    className : 'Label',
    theme     : 'light',
    disabled  : false
};

module.exports = Label;
