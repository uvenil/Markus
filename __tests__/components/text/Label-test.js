'use strict';

import React from 'react';
import Renderer from 'react-test-renderer';
import Label from '../../../app/components/text/Label.jsx';

it('renders correctly', () => {
    const label = Renderer.create(
        <Label />
    ).toJSON();

    expect(label).toMatchSnapshot();
});
