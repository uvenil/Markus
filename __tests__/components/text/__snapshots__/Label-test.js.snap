exports[`test renders correctly 1`] = `
<span
  className="Label"
  style={
    Object {
      "WebkitUserSelect": "none",
      "color": "#000",
      "cursor": "default",
      "fontFamily": undefined,
      "fontSize": undefined,
      "fontWeight": undefined,
      "pointerEvents": "auto",
    }
  } />
`;
